compiler/hash-source/src/constant.rs

```diff
@@ -0,0 +1,291 @@
+//! Hash source interned constant store. This is a data store
+//! for interning observed constants within the source file of
+//! a program. This is done so that constants don't need to
+//! be passed around each stage of the compiler and can just
+//! be interned, and accessed when needed.
+
+use std::fmt::Display;
+
+use dashmap::DashMap;
+use fnv::FnvBuildHasher;
+use hash_utils::counter;
+use lazy_static::lazy_static;
+use num_bigint::BigInt;
+
+use crate::identifier::{Identifier, CORE_IDENTIFIERS};
+
+/// Interned float constant which stores the value of the float, and
+/// an optional `type ascription` which is a suffix on the literal
+/// describing which float kind it is, either being `f32` or `f64`.
+#[derive(Debug, Clone, Copy)]
+pub struct FloatConstant {
+    /// Raw value of the float
+    pub value: f64,
```

Comment:
Would it be valid to have NaN or Infinity here ?

Reply:
I believe that `NaN` and `inf` can be represented within `f64`, so yes?